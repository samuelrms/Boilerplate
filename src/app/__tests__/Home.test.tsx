import { renderWithTheme } from '@/tests/utils';
import { screen } from '@testing-library/react';
import Home from '../page';

describe('Home', () => {
  test('should render the heading', () => {
    renderWithTheme(<Home />);
    expect(
      screen.getByRole('heading', {
        name: /Create Next.js apps using TailwindCSS and TypeScript/i,
      })
    );
  });

  test('should render the Typography component', () => {
    renderWithTheme(<Home />);
    expect(screen.getByText(/Typography component/i));
  });

  test('should render the Button component', () => {
    renderWithTheme(<Home />);
    expect(screen.getByText(/Button Component Primary/i));
  });

  test('should render the Button component with secondary color', () => {
    renderWithTheme(<Home />);
    expect(screen.getByText(/Button Component Secondary/i));
  });

  test('should render the Stack component', () => {
    renderWithTheme(<Home />);
    expect(screen.getByText(/Stack component/i));
  });

  test('should render the Divider component', () => {
    renderWithTheme(<Home />);
    expect(screen.getAllByRole('separator').length).toBe(2);
  });

  test('should render the main tag', () => {
    renderWithTheme(<Home />);
    expect(screen.getByRole('main'));
  });

  test('should render correctly', () => {
    const { container } = renderWithTheme(<Home />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
