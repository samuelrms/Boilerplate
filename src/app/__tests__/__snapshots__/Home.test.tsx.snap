// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render correctly 1`] = `
<main
  class="flex min-h-screen items-center justify-center p-24 flex-col"
>
  <h1
    class="text-4xl break-words text-center"
  >
    Create Next.js apps using TailwindCSS and TypeScript
  </h1>
  <h3
    class="css-1ndg5q"
  >
    Typography component
  </h3>
  <hr
    class="css-1appgvr"
    height="3"
  />
  <button
    class="css-euufcq"
  >
    Button Component Primary
  </button>
  <hr
    class="css-1appgvr"
    height="3"
  />
  <button
    class="css-a0h23x"
    color="secondary"
  >
    Button Component Secondary
  </button>
  <div
    class="css-1botpz1"
    color="#f6f6f6"
  >
    Stack component
  </div>
</main>
`;
